Sources/SemanticAnalyzer/SemanticAnalyzer.swift

```diff
@@ -70,10 +70,41 @@ public struct SemanticAnalyzer: ASTPass {
 
     return ASTPassResult(element: structDeclaration, diagnostics: diagnostics, passContext: passContext)
   }
-
+  
+  public func process(enumDeclaration: EnumDeclaration, passContext: ASTPassContext) -> ASTPassResult<EnumDeclaration> {
+    var diagnostics = [Diagnostic]()
+    
+    if let conflict = passContext.environment!.conflictingTypeDeclaration(for: enumDeclaration.identifier) {
+      diagnostics.append(.invalidRedeclaration(enumDeclaration.identifier, originalSource: conflict))
+    }
+    
+    if case .basicType(_) = enumDeclaration.typeAnnotation.type.rawType {} else {
```

Comment:
this is confusing - could you at least put a newline in between the braces here?